core/src/main/java/org/geysermc/geyser/translator/protocol/java/level/JavaLevelEventTranslator.java

```diff
@@ -346,6 +367,35 @@ public void translate(GeyserSession session, ClientboundLevelEventPacket packet)
                 session.sendUpstreamPacket(soundEventPacket);
                 return;
             }
+            // Half of these are listed in MCProtocolLib as Trail instead of Trial for some reason
+            case PARTICLES_TRIAL_SPAWNER_DETECT_PLAYER -> {
+                // Particles spawn here
+                TrialSpawnerDetectEventData eventData = (TrialSpawnerDetectEventData) packet.getData();
+                effectPacket.setType(org.cloudburstmc.protocol.bedrock.data.LevelEvent.PARTICLE_TRAIL_SPAWNER_DETECTION);
+                // 1.0 is used here for Y instead of 0.5 to match Java Positioning.
+                // 0.5 is what the BDS uses for positioning.
+                effectPacket.setPosition(pos.sub(0.5f, 1.0f, 0.5f));
+                effectPacket.setData(eventData.getDetectedPlayers());
+            }
+            case PARTICLES_TRIAL_SPAWNER_DETECT_PLAYER_OMINOUS -> {
+                /*
+                    Particles dont spawn here for some reason, only sound plays
+                    This seems to be a bug in v1.21.0 and v1.21.1: see https://bugs.mojang.com/browse/MCPE-181465
+                    If this gets fixed, the positioning should be the same as normal activation.
+                   */
+                effectPacket.setType(org.cloudburstmc.protocol.bedrock.data.LevelEvent.PARTICLE_TRIAL_SPAWNER_DETECTION_CHARGED);
+                effectPacket.setPosition(pos.sub(0.5f, 1.0f, 0.5f));
+            }
+            case PARTICLES_TRIAL_SPAWNER_BECOME_OMINOUS -> effectPacket.setType(org.cloudburstmc.protocol.bedrock.data.LevelEvent.PARTICLE_TRIAL_SPAWNER_BECOME_CHARGED);
+            case PARTICLES_TRIAL_SPAWNER_SPAWN, PARTICLES_TRIAL_SPAWNER_SPAWN_MOB_AT -> {
+                // This should be its own class in MCProtocolLib.
+                // if 0, use Orange Flames,
+                // if 1, use Blue Flames for ominous spawners
```

Comment:
Did you check which level events are sent by the Bedrock server (using proxypass + vanilla bedrock server, also called BDS from here: https://www.minecraft.net/en-us/download/server/bedrock)? Might be that the level event names are just different

Reply:
Yes I did, it uses the same event.

Reply:
Ah, seems like a Bedrock limitation then. Thanks for checking!